Week1/homework/app.js

```diff
@@ -1,11 +1,140 @@
 'use strict';
 {
+  //2.1
   const bookTitles = [
-    // Replace with your own book titles
-    'harry_potter_chamber_secrets'
+    'brain_camp ',
+    'sophie_washington_secret_santa',
+    'el_deafo',
+    'american_born_chinese',
+    'queen_of_the_world!',
+    'land_of_sad_oranges',
+    'return-to-haifa',
+    'in-the-presence-of-absence',
+    'black-great-on-you',
+    'do-not-be-sad',
   ];
+  //2.3
+  function bodyFunction() {
+    const main_list = document.getElementById('main_list');
+    const h1 = document.createElement('h1');
+    h1.innerText = 'My Books:';
+    main_list.appendChild(h1);
+    const ul = document.createElement('ul');
+    main_list.appendChild(ul);
+    for (let bookTitlesIndex = 0; bookTitlesIndex < bookTitles.length; bookTitlesIndex++) {
+      const li = document.createElement("li");
+      ul.appendChild(li);
+      li.innerText = bookTitles[bookTitlesIndex];
+    }
+  }
+  window.addEventListener('load', bodyFunction);
+
+  function bookTitlesList(ul) {
+    for (let bookTitlesIndex = 0; bookTitlesIndex < bookTitles.length; bookTitlesIndex++) {
+      const li = document.createElement("li");
+      ul.appendChild(li);
+      li.innerText = bookTitles[bookTitlesIndex];
+    }
+  }
+
+  //2.4
+  const bookDetails = {
+    'brain_camp': { title: 'Brain Camp', language: 'English', author: 'Susan Kim' },
+    'sophie_washington_secret_santa': { title: 'Sophie Washington Secret Santa', language: 'English', author: 'Tonya Duncan Ellis ' },
+    'el_deafo': { title: 'El Deafo', language: 'English', author: 'Cece Bell' },
+    'american_born_chinese': { title: 'American Born Chinese', language: 'English', author: 'Gene Luen Yang' },
+    'queen_of_the_world': { title: 'Queen of the World!', language: 'English', author: 'Jennifer L. Holm' },
+    'land_of_sad_oranges': { title: 'Land of sad oranges', language: 'Arabic', author: 'Ghassan Kanafani' },
+    'return_to_haifa': { title: 'Return to Haifa', language: 'Arabic', author: 'Ghassan Kanafani' },
+    'in_the_presence_of_absence': { title: 'In The Presence Of Absence', language: 'Arabic', author: 'Mahmoud Darwish' },
+    'black_great_on_you': { title: 'Black great on you', language: 'Arabic', author: 'Ahlam mustghanmy' },
+    'do_not_be_sad': { title: 'Don\'t Be Sad', language: 'Arabic', author: 'Alqarny' },
+  };
+
+
+  //2.5
+  function bookDetailsFunction() {
+    const main_list = document.getElementById('main_list');
+    const h1 = document.createElement('h1');
+    h1.innerText = 'My Books:';
+    main_list.appendChild(h1);
+    const ul = document.createElement('ul');
+    main_list.appendChild(ul);
+    const bookDetailsKeys = Object.keys(bookDetails);
+    for (let bookDetailsIndex = 0; bookDetailsIndex < bookDetailsKeys.length; bookDetailsIndex++) {
```

Comment:
Instead of getting the keys from the object, it is better to use the titles from the `bookTitles` array, as that array was used to create the initial `li` elements.

So, the `bookDetails` object defines which books are available and the `bookTitles` array defines which books to display and in which order to display them.